doc/releases/changelog-dev.md

```diff
@@ -256,6 +256,39 @@
   output of `qml.specs()`. 
   [(#4015)](https://github.com/PennyLaneAI/pennylane/pull/4015)
 
+* The `ResourcesOperation` class is added to allow users to define operations with custom resource information.
+  [(#4026)](https://github.com/PennyLaneAI/pennylane/pull/4026)
+
+  For example, users can define a custom operation by inheriting from this new class:
+
+  ```pycon
+  >>> class CustomOp(ResourcesOperation):
+  ...     def resources(self):
+  ...         return Resources(num_wires=1, num_gates=2, gate_types={"PauliX": 2})
+  ... 
+  >>> CustomOp(wires=1)
+  CustomOp(wires=[1])
+  ```
+  
+  Then we can track and display the resources of the workflow using `tape.specs`:
```

Comment:
Minor point: should we use an example with `qml.specs`?

Reply:
Added 👍🏼